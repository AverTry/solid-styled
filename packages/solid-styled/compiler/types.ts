import { Scope } from '@babel/traverse';
import * as t from '@babel/types';
import postcssrc from 'postcss-load-config';

export interface SolidStyledOptions {
  verbose?: boolean;
  prefix?: string;
  env?: 'development' | 'production';
}

export interface ScopedSheet {
  id: t.Identifier;
  scope: string;
  count: number;
}

export interface StateContext {
  hooks: Map<string, t.Identifier>;
  vars: WeakMap<Scope, t.Identifier>;
  sheets: WeakMap<Scope, ScopedSheet>;
  opts: SolidStyledOptions;
  ids: number;
  ns: string;
  postcss: postcssrc.Result;
}
