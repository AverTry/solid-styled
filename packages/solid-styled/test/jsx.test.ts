import { describe, expect, it } from 'vitest';
import type { SolidStyledOptions } from '../compiler';
import { compile } from '../compiler';

const options: SolidStyledOptions = {
  verbose: false,
};

const FILE = 'src/index.ts';

describe('jsx', () => {
  it('should transform', async () => {
    const code = `
export default function Example() {
  return (
    <>
      <style jsx>
        {\`
          h1 {
            color: red;
          }
        \`}
      </style>
      <h1>Hello World</h1>
    </>
  );
}
  `;
    expect((await compile(FILE, code, options)).code).toMatchSnapshot();
  });
  it('should work with multiple templates', async () => {
    const code = `
export default function Example() {
  return (
    <>
      <style jsx>
        {\`
          h1 {
            color: red;
          }
        \`}
      </style>
      <h1>Hello World</h1>
      <style jsx>
        {\`
          h2 {
            color: blue;
          }
        \`}
      </style>
      <h2>Hello World</h2>
    </>
  );
}
  `;
    expect((await compile(FILE, code, options)).code).toMatchSnapshot();
  });
  it('should transform dynamic templates', async () => {
    const code = `
    export default function Example() {
      return (
        <>
          <style jsx>
            {\`
              h1 {
                color: \${props.color};
              }
            \`}
          </style>
          <h1>Hello World</h1>
        </>
      );
    }
  `;
    expect((await compile(FILE, code, options)).code).toMatchSnapshot();
  });
});
