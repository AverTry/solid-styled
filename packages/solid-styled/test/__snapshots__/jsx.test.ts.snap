// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`jsx > should transform 1`] = `
"import { useSolidStyled as _useSolidStyled } from \\"solid-styled\\";
const _sheet = \\"c-f0b3b6fa-0\\";
const _css = \\"h1[s\\\\\\\\:c-f0b3b6fa-0]{color:red}\\";
export default function Example() {
  _useSolidStyled(_sheet, 1, _css);
  return <>
      
      <h1 s:c-f0b3b6fa-0>Hello World</h1>
    </>;
}"
`;

exports[`jsx > should transform dynamic templates 1`] = `
"import { useSolidStyled as _useSolidStyled } from \\"solid-styled\\";
import { createCSSVars as _createCSSVars } from \\"solid-styled\\";
const _sheet = \\"c-f0b3b6fa-0\\";
const _css = \\"h1[s\\\\\\\\:c-f0b3b6fa-0]{color:var(--s-v-f0b3b6fa-1)}\\";
export default function Example() {
  const _vars = _createCSSVars();
  _useSolidStyled(_sheet, 1, _css), _vars(() => ({
    \\"--s-v-f0b3b6fa-1\\": props.color
  }));
  return <>
          
          <h1 s:c-f0b3b6fa-0 style={_vars()}>Hello World</h1>
        </>;
}"
`;

exports[`jsx > should work with multiple templates 1`] = `
"import { useSolidStyled as _useSolidStyled } from \\"solid-styled\\";
const _sheet = \\"c-f0b3b6fa-0\\";
const _css = \\"h1[s\\\\\\\\:c-f0b3b6fa-0]{color:red}\\";
const _css2 = \\"h2[s\\\\\\\\:c-f0b3b6fa-0]{color:#00f}\\";
export default function Example() {
  _useSolidStyled(_sheet, 1, _css);
  _useSolidStyled(_sheet, 2, _css2);
  return <>
      
      <h1 s:c-f0b3b6fa-0>Hello World</h1>
      
      <h2 s:c-f0b3b6fa-0>Hello World</h2>
    </>;
}"
`;

exports[`jsx.global > should transform 1`] = `
"import { useSolidStyledGlobal as _useSolidStyledGlobal } from \\"solid-styled\\";
const _sheet = \\"c-f0b3b6fa-0\\";
const _css = \\"h1{color:red}\\";
export default function Example() {
  _useSolidStyledGlobal(_sheet, 1, _css);
  return <>
      
      <h1 s:c-f0b3b6fa-0>Hello World</h1>
    </>;
}"
`;

exports[`jsx.global > should transform dynamic templates 1`] = `
"import { useSolidStyledGlobal as _useSolidStyledGlobal } from \\"solid-styled\\";
import { createCSSVars as _createCSSVars } from \\"solid-styled\\";
const _sheet = \\"c-f0b3b6fa-0\\";
const _css = \\"h1{color:var(--s-v-f0b3b6fa-1)}\\";
export default function Example() {
  const _vars = _createCSSVars();
  _useSolidStyledGlobal(_sheet, 1, _css, _vars(() => ({
    \\"--s-v-f0b3b6fa-1\\": props.color
  })));
  return <>
          
          <h1 s:c-f0b3b6fa-0 style={_vars()}>Hello World</h1>
        </>;
}"
`;

exports[`jsx.global > should work with multiple templates 1`] = `
"import { useSolidStyledGlobal as _useSolidStyledGlobal } from \\"solid-styled\\";
const _sheet = \\"c-f0b3b6fa-0\\";
const _css = \\"h1{color:red}\\";
const _css2 = \\"h2{color:#00f}\\";
export default function Example() {
  _useSolidStyledGlobal(_sheet, 1, _css);
  _useSolidStyledGlobal(_sheet, 2, _css2);
  return <>
      
      <h1 s:c-f0b3b6fa-0>Hello World</h1>
      
      <h2 s:c-f0b3b6fa-0>Hello World</h2>
    </>;
}"
`;
